Sources/ScheduledTasks.php

```diff
@@ -1542,4 +1542,196 @@ function scheduled_remove_old_drafts()
+
+	// Find the oldest $maxMembers in log_boards
+	$sql = "SELECT lb.id_member, m.last_login
+		FROM {db_prefix}members m
+		INNER JOIN
+		(
+			SELECT DISTINCT id_member
+			FROM {db_prefix}log_boards
+		) lb
+		ON lb.id_member = m.id_member
+		WHERE m.last_login <= {int:cutoff}
+		ORDER BY m.last_login
+		LIMIT {int:limit}";
+	$result = $smcFunc['db_query']('', $sql,
+		array(
+			'limit' => $maxMembers,
+			'cutoff' => $cleanupBeyond,
+		)
+	);
+	$members = array();
+	while($row = $smcFunc['db_fetch_assoc']($result))
+		$members[$row['id_member']] = $row['last_login'];
+	$smcFunc['db_free_result']($result);
+
+	// Find the oldest $maxMembers in log_mark_read
+	$sql = "SELECT lmr.id_member, m.last_login
+		FROM {db_prefix}members m
+		INNER JOIN
+		(
+			SELECT DISTINCT id_member
+			FROM {db_prefix}log_mark_read
+		) lmr
+		ON lmr.id_member = m.id_member
+		WHERE m.last_login <= {int:cutoff}
+		ORDER BY m.last_login
+		LIMIT {int:limit}";
+	$result = $smcFunc['db_query']('', $sql,
+		array(
+			'limit' => $maxMembers,
+			'cutoff' => $cleanupBeyond,
+		)
+	);
+	// Using id_member as key basically eats dupes...
+	while($row = $smcFunc['db_fetch_assoc']($result))
+		$members[$row['id_member']] = $row['last_login'];
+	$smcFunc['db_free_result']($result);
+
+	// Find the oldest $maxMembers in log_topics
+	// Note the different cutoff, as we may need to mark these as read
+	$sql = "SELECT lt.id_member, m.last_login
+		FROM {db_prefix}members m
+		INNER JOIN
+		(
+			SELECT DISTINCT id_member
+			FROM {db_prefix}log_topics
+			WHERE unwatched = {int:unwatched}
+		) lt
+		ON lt.id_member = m.id_member
+		WHERE m.last_login <= {int:cutoff}
+		ORDER BY m.last_login
+		LIMIT {int:limit}";
+	$result = $smcFunc['db_query']('', $sql,
+		array(
+			'limit' => $maxMembers,
+			'cutoff' => $markReadCutoff,
+			'unwatched' => 0,
+		)
+	);
+	// Using id_member as key basically eats dupes...
+	while($row = $smcFunc['db_fetch_assoc']($result))
+		$members[$row['id_member']] = $row['last_login'];
+	$smcFunc['db_free_result']($result);
+
+	// Sort by last_login & limit to $maxMembers entries
+	asort($members);
+	$members = array_slice($members, 0, $maxMembers, TRUE);
+
+	// Either mark boards as read or purge entries, based on last_login
+	foreach($members as $id => $last_login)
+	{
+		if ($last_login <= $cleanupBeyond)
+		{
+			// Not active in a while - just purge the entries
+
+			// Delete rows from log_boards
+			$sql = "DELETE FROM {db_prefix}log_boards
+				WHERE id_member = {int:member}";
+			$smcFunc['db_query']('', $sql,
+				array(
+					'member' => $id,
+				)
+			);
+			// Delete rows from log_mark_read
+			$sql = "DELETE FROM {db_prefix}log_mark_read
+				WHERE id_member = {int:member}";
+			$smcFunc['db_query']('', $sql,
+				array(
+					'member' => $id,
+				)
+			);
+			// Delete rows from log_topics
+			$sql = "DELETE FROM {db_prefix}log_topics
+				WHERE id_member = {int:member}
+				AND unwatched = {int:unwatched}";
```

Comment:
move the line one tabe to right,
since "and" is part of the where

Reply:
Will do...